### PR TITLE
wip: add description of features

### PR DESCRIPTION
**Pull request by cstml**

### CODE REVIEW

File: docs/comparison-matrix.md [hunk 1]
```diff
@@ -31,8 +31,72 @@ Legend:
...
+
+Recursive types are defined by the presence of the LHS type in its RHS
+definition. A classic example is:
+
+```text
+List a = Nil | Cons a (List a)
+^^^^^^                 ^^^^^^
+```
+
+### Type Functions (Generics)
```
Comment (bladyjoker) [inline docs/comparison-matrix.md:62]: Perhaps it's better what @gnumonik suggested. `Parameterized Types (Generics)`?

File: docs/comparison-matrix.md [hunk 2]
```diff
@@ -31,8 +31,72 @@ Legend:
...
+
+Type functions allow for the introduction of type variables in the LHS definition
+of the term - creating a parametrised type definition. The classic example is
```
Comment (bladyjoker) [inline docs/comparison-matrix.md:65]: typo; should be parameterised? or even parameterized?

File: docs/comparison-matrix.md [hunk 3]
```diff
@@ -31,8 +31,72 @@ Legend:
...
+```text
+Maybe a = Nothing | Just a
+```
+
+Using the above type definition we can now define another type that uses `Maybe`
+and instantiates it to use `Integer`
+
+```text
+Time_Saved_via_LambdaBuffers = Maybe Integer
+```
```
Comment (bladyjoker) [inline docs/comparison-matrix.md:77]: lol

File: docs/comparison-matrix.md [hunk 4]
```diff
@@ -31,8 +31,72 @@ Legend:
...
+
+### Type Annotations / Constraints
+
+:FIXME: @Drazen
+
+### Add New Built-in Types
+
+Refer to [./design](./design.md) - section `### Extensibility to new types`.
```
Comment (bladyjoker) [inline docs/comparison-matrix.md:85]:
```
Refer to the [Design](design.md#extensible-to-new-types).
```

File: docs/comparison-matrix.md [hunk 5]
```diff
@@ -31,8 +31,72 @@ Legend:
...
+
+### Add New Type Semantics
+
+Refer to [./design](./design.md) - section `### Extensibility to new semantics`.
```
Comment (bladyjoker) [inline docs/comparison-matrix.md:89]:
```
Refer to the [Design](design.md#extensible-to-new-semantics).
```

File: docs/comparison-matrix.md [hunk 6]
```diff
@@ -31,8 +31,72 @@ Legend:
...
+
+### Manage type semantics (at language level)
+
+Refer to [./design](./design.md) - section `### Extensibility to new semantics`.
```
Comment (bladyjoker) [inline docs/comparison-matrix.md:93]:
```
Refer to the [Design](design.md#expressive-semantics-annotation).
```

File: docs/comparison-matrix.md (hunk 1 above)
Comment (cstml) [reply docs/comparison-matrix.md:61]:
```suggestion
### Parameterized Types (Generics)
```

File: docs/comparison-matrix.md [hunk 7]
```diff
@@ -13,8 +13,8 @@ Legend:
...
 |--------------------------------------------------------|-------------------|--------------|-----------------|--------------------|----------|--------------|
 | Sum Types                                              | 🟢                | 🟢           | 🔴              | 🟢                 | 🟢       | 🟢           |
-| Record Types                                           | 🟢                | 🟢           | 🟢              | 🟢                 | 🟢       | 🟢           |
 | Product Types                                          | 🔴                | 🔴           | 🔴              | 🟢                 | ❔       | 🔴           |
+| Record Types                                           | 🟢                | 🟢           | 🟢              | 🟢                 | 🟢       | 🟢           |
 | Recursive Types                                        | 🟢                | 🟢           | 🔴              | 🟢                 | 🟢       | ❔           |
 | Type functions (Generics)                              | 🔴                | 🟢           | 🔴              | 🟢                 | 🟢       | 🔴           |
```
Comment (cstml) [inline docs/comparison-matrix.md:19]:
```suggestion
| Parameterized Types (Generics)                              | 🔴                | 🟢           | 🔴              | 🟢                 | 🟢       | 🔴           |
```

File: docs/comparison-matrix.md (hunk 4 above)
Comment (cstml) [reply docs/comparison-matrix.md:85]:
```suggestion
Refer to [design document](design.md#extensible-to-new-types).
```

File: docs/comparison-matrix.md (hunk 5 above)
Comment (cstml) [reply docs/comparison-matrix.md:89]:
```suggestion
Refer to the [design document](design.md#extensible-to-new-semantics).
```

File: docs/comparison-matrix.md [hunk 8]
```diff
@@ -23,16 +23,115 @@ Legend:
...
+
+Refer to [./design](./design.md) - section `### Extensibility to new types`.
+
+### Add New Type Semantics
+
+Refer to [./design](./design.md) - section `### Extensibility to new semantics`.
+
+### Manage Type Semantics (at Language Level)
+
+Refer to [./design](./design.md) - section `### Extensibility to new semantics`.
```
Comment (cstml) [inline docs/comparison-matrix.md:93]:
```suggestion
Refer to the [design document](design.md#expressive-semantics-annotation)..
```

File: docs/comparison-matrix.md (hunk 6 above)
Comment (cstml) [reply docs/comparison-matrix.md:93]:
```suggestion
Refer to the [design document](design.md#expressive-semantics-annotation).
```

**Pull request merged**